### PR TITLE
Added parameters validator for DatabaseInitializationJob

### PR DESCRIPTION
**Pull request by junaidnz97**
The purpose of this PR is to add Parameter validation and associated tests to the job DatabaseInitialization

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidator.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public void validate(JobParameters parameters) throws JobParametersInvalidException {
+        compositeJobParametersValidator(parameters).validate(parameters);
+    }
+
+    private CompositeJobParametersValidator compositeJobParametersValidator(JobParameters jobParameters) {
+        List<JobParametersValidator> jobParametersValidators = new ArrayList<>();
+
+        jobParametersValidators.add(new CreateDatabaseIndexesStepParametersValidator());
+        jobParametersValidators.add(new LoadGenesStepParametersValidator());
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidator.java]: Given that we are not loading just genes but also transcripts, and IIRC the stored object is called `FeatureCoordinates`, I suggest renaming this class to `LoadFeatureCoordinatesStepParameteresValidator`.

Related classes such as the step itself may also need to be renamed accordingly.

File: src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private Map<String, JobParameter> optionalParameters;
+
+    @Before
+    public void setUp() throws Exception {
+        validator = new DatabaseInitializationJobParametersValidator();
+
+        requiredParameters = new TreeMap<>();
+        requiredParameters.put(JobParametersNames.DB_NAME, new JobParameter(DATABASE_NAME));
+        requiredParameters.put(JobParametersNames.DB_COLLECTIONS_FEATURES_NAME,
+                new JobParameter(COLLECTION_FEATURES_NAME));
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidatorTest.java]: To follow our coding style guidelines:
```suggestion
                               new JobParameter(COLLECTION_FEATURES_NAME));
```

File: src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Test
+    public void allRequiredJobParametersAreValid() throws JobParametersInvalidException {
+        Map<String, JobParameter> parameters = new TreeMap<>();
+        parameters.putAll(requiredParameters);
+        validator.validate(new JobParameters(parameters));
+    }
+
+    @Test(expected = JobParametersInvalidException.class)
+    public void dbNameIsRequired() throws JobParametersInvalidException {
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidatorTest.java]: Please create one test method per required parameter.

Comment (junaidnz97) [reply src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidatorTest.java]: Added 👍

File: src/main/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidator.java (hunk 1 above)
Comment (junaidnz97) [reply src/main/java/uk/ac/ebi/eva/pipeline/parameters/validation/job/DatabaseInitializationJobParametersValidator.java]: Changed :)

**Pull request merged**